src/legacy/core_plugins/expressions/public/np_ready/public/loader.ts

```diff
@@ -88,11 +100,19 @@ export class ExpressionLoader {
     return this.dataHandler.inspect();
   }
 
-  update(expression: string | ExpressionAST, params: IExpressionLoaderParams): Promise<RenderId> {
+  update(expression?: string | ExpressionAST, params?: IExpressionLoaderParams): Promise<RenderId> {
     const promise = this.render$.pipe(first()).toPromise();
+    if (params && params.searchContext && this.params.searchContext) {
+      this.params.searchContext = _.defaults(
+        {},
+        params.searchContext,
+        this.params.searchContext
+      ) as any;
+    }
 
-    if (expression !== null) {
-      this.execute(expression, params);
+    this.loadingSubject.next();
```

Comment:
Does it make sense to trigger a loading state for both `update` and `render` cases? Suggestion: remove this line from the `update` function

Reply:
we want to emit `loading` whenever we are re-rendering as well. render is async function and might take some time to finish. that's why one reason we emit loading in this case as well. The second reason is that after visualization rerendered it might look completely different than it did before, even if we did not refetch data, and that's important for reporting and functional tests to know.

Reply:
This is fine, I will need to modify my PR to account for this